diff -ruN a/gstreamer_ti/ti_build/ticodecplugin/src/gsttithreadprops.h b/gstreamer_ti/ti_build/ticodecplugin/src/gsttithreadprops.h
--- a/gstreamer_ti/ti_build/ticodecplugin/src/gsttithreadprops.h	2010-01-27 21:52:56.022218358 +0300
+++ b/gstreamer_ti/ti_build/ticodecplugin/src/gsttithreadprops.h	2010-01-27 21:56:55.074129258 +0300
@@ -41,7 +41,7 @@
 
 /* Thread status utility macros. */
 #define gst_tithread_lock_status(x) \
-            pthread_mutex_lock(&(x)->threadStatusMutex)
+            pthread_mutex_trylock(&(x)->threadStatusMutex)
 #define gst_tithread_unlock_status(x) \
             pthread_mutex_unlock(&(x)->threadStatusMutex)
         
